lib/literal/data_structure.rb

```diff
@@ -11,13 +11,29 @@ def self.from_pack(payload)
 	end
 
 	def [](key)
-		instance_variable_get(:"@#{key}")
+		case key
+		when Symbol
+		when String
+			key = key.intern
```

Comment:
Im personally not a fan of 'indifferent access' on structures etc. Could this be opt-in maybe? Or another specific data structure base class ala "data structure with indifferent access"?

Reply:
I was mirroring how the stdlib Struct/Data work

Reply:
I don’t feel strongly about this one way or the other. I wouldn’t add indifferent access to Literal::Hash, but I think it’s okay here.

Reply:
For Struct I can see the appeal of mirroring stdlib. Data however doesn't implement [] if I'm not mistaken, maybe we could just add this to Struct?